scribble-lib/scribble/blueboxes.rkt

```diff
@@ -62,30 +88,42 @@
             (apply
              append
              (for/list ([offset+len (in-list offset+lens)])
+               : (Listof (Listof (U String EOF)))
                (call-with-input-file blueboxes.rktd
-                 (λ (port)
+                 (λ ([port : Input-Port])
                    (port-count-lines! port)
-                   (file-position port (+ (car offset+len) offset))
+                   (file-position port (+ (car offset+len) (or offset 0)))
                    (for/list ([i (in-range (cdr offset+len))])
+                     : (Listof (U String EOF))
                      (read-line port)))))))
           (cond
             [(ormap eof-object? lines) #f]
-            [else lines])]
-         [else #f])])))
+            [(null? lines) #f]
+            [else (assert (filter string? lines) pair?)])]
```

Comment:
On Sun, Jun 28, 2015 at 4:20 PM, Alex Knauth <notifications@github.com>
wrote:

> In scribble-lib/scribble/blueboxes.rkt
> <https://github.com/racket/scribble/pull/8#discussion_r33427136>:
>
> >                       (read-line port)))))))
> >            (cond
> >              [(ormap eof-object? lines) #f]
> > -            [else lines])]
> > -         [else #f])])))
> > +            [(null? lines) #f]
> > +            [else (assert (filter string? lines) pair?)])]
>
> Yes, but since the fetch-strs-for-single-tag function is supposed to
> return either a non-empty list or #f, the previous behavior was a bit
> wrong (according to the contract on fetch-blueboxes-strs that was there
> before).
>
>

Cool!! :)